proto/p4/config/p4info.proto

```diff
@@ -55,6 +53,22 @@ message Preamble {
   repeated string annotations = 4;
 }
 
+// used to group all extern instances of the same type in one message
+message Extern {
+  // the extern_type_id is assigned during compilation. It is likely that this
+  // id will in fact come from a P4 annotation to the extern declaration and
+  // that each vendor will receive a prefix to avoid collisions.
```

Comment:
An alternative we could consider would be to generate the id as a hash of the name. The likelihood of a collision is pretty low when considering how many types of P4 externs are likely to be used within the same program.